README.md

```diff
@@ -76,317 +110,607 @@ cartridge --help
+    * click **Create** in the web interface;
+
+  * join the instance to an existing cluster once you:
+
+    * run `cartridge.join_server({uri = 'other_instance_uri'})` via the console, or
+    * click **Join** (an existing replica set) or **Create** (a new replica set)
+      in the web interface.
+
+Notice that you can specify a cookie for the cluster (`cluster_cookie` parameter)
+if you need to run several clusters in the same network. The cookie can be any
+string value.
+
+Now you can develop an application that will run on a single or multiple
+independent Tarantool instances (e.g. acting as a proxy to third-party databases)
+-- or will run in a cluster.
+
+  **NOTE:** If you plan to develop a cluster-aware application, first familiarize
+  yourself with the notion of
+  [cluster roles](https://www.tarantool.io/en/doc/latest/book/cartridge/cartridge_dev/#cluster-roles).
+
+### Building an application
+
+#### Building locally
+
+To build your application locally (for local testing), say this in any directory:
 
 ```sh
-cartridge build ./myapp
+cartridge build [<path>]
 ```
 
-Run instances locally:
+This command requires one argument -- the path to your application directory
+(i.e. to the build source). The default path is `.` (the current directory).
+
+This command runs:
+
+1. `cartridge.pre-build` (or [DEPRECATED] `.cartridge.pre`), if the
+   [pre-build file](#special-files) exists.
+   This builds the application in the `path` directory.
+2. `tarantoolctl rocks make`, if the [rockspec file](#special-files) exists.
+   This installs all Lua rocks to the `path` directory.
+
+During step 1, `cartridge` creates a temporary build directory (by default,
+`~/.cartridge/tmp/`) where application build is done. You can
+[change](#build-directory) the default build directory, if needed.
+
+During step 2 -- the key step here -- `cartridge` installs all dependencies
+specified in the rockspec file (you can find this file within the application
+directory created from template).
+
+  **NOTE:** An advanced alternative would be to specify build logic in the
+  rockspec as `cmake` commands, like we
+  [do it](https://github.com/tarantool/cartridge/blob/master/cartridge-scm-1.rockspec#L26).
+  for `cartridge`.
+
+If your application depends on closed-source rocks, or if the build should contain
+rocks from a project added as a submodule, then you need to **install** all these
+dependencies before calling `tarantoolctl rocks make`.
+You can do it using the file `cartridge.pre-build` in your application root
+(again, you can find this file within the application directory created from template).
+In this file, you can specify all rocks to build
+(e.g. `tarantoolctl rocks make --chdir ./third_party/proj`).
+For details, see [special files](#special-files).
+
+As a result, in the application's `.rocks` directory you will get a fully built
+application that you can start locally from the application's directory.
+
+#### Building in Docker
```

Comment:
The `cartridge build` command is used only for local development, so there is no option to build in docker.